File: languages/java/src/provider/MeanAnalysis.java
package provider;

import model.SimpleTableAnalysis;
import model.UserInfo;

import java.util.List;

public class MeanAnalysis implements SimpleTableAnalysis {

    @Override
    public double analysis(List<UserInfo> userInfoList) {

        double sum = 0;
        for(UserInfo userInfo : userInfoList){
            sum += userInfo.getCredit();
        }
        return sum/userInfoList.size();
    }
}

File: Languages/R/src/provider/MeanAnalysis.R
# Criando uma classe "MeanValueAnalysis"
MeanAnalysis = R6Class("MeanAnalysis",
                   public = list(
                     
                     # Método responsável por obter a media dos valore de credit na lista de usuarios
                     # @param userInfoList Lista de usuarios
                     # @return Media de valores de credit da lista
                     analysis = function(userInfoList) {
                       creditVector = userInfoList$getCredit()
                       creditVector = as.numeric(creditVector)
                       return(mean(creditVector))
                     }
                   )
)